examples/protocol_specification_ex/gym.yaml

```diff
@@ -22,3 +22,12 @@ speech_acts:
 ct:AnyObject: |
   bytes any = 1;
 ...
+---
+reply:
+  act: [percept]
+  percept: []
+  reset: []
+  close: []
+roles: {agent, environment}
+end_states: [open, closed]
```

Comment:
only `closed` in line with above

Reply:
I am not quite sure if I understood the gym interaction here. Is it that it allows three dialogue streams, the first being maximum two messages and the others single message dialogues:

`act, percept`
`reset`
`close`

If so, I suggest:
`end_states: [act-percept_loop, reset, close]`

Reply:
We could narrow it further:
```
  act: [percept]
  percept: [act, close, reset]
  close: []
  reset: []
```
End states should be only `closed` I suppose.

Reply:
Alternatively, `success` or `finished`..`act_percept_exchange` is very verbose and not very informative.

Reply:
Ok with the new reply structure, there are two ways a dialogue may end, either with `close` or `reset`, so the `end_states: [reset, closed]`.